lib/kuzzle/kuzzle.ts

```diff
@@ -307,118 +539,47 @@ class Kuzzle extends KuzzleEventEmitter {
    *
    * @returns {Promise<void>}
    */
-  async import (toImport = {}, toSupport = {}) {
-    const types = ['userMappings', 'mappings', 'fixtures', 'permissions'];
-    // If a mutex fail to acquire the lock on the first attempt,
-    // it means that another node has taken care of doing so...
-    // Therefore, there is no need to do the same import twice.
-    const mutexes = types.reduce((result, type) => {
-      result[type] = new Mutex(`backend:import:${type}`, { timeout: 0 });
-      return result;
-    }, {});
-    // Following the same logic as before, mutexes are locked longer
-    // than necessary in order to rarely execute the same import
-    const lockedMutexes = [];
-
-    this.log.info('[ℹ] Imports in progress: This can take some time.');
+  async import (
+    toImport: ImportConfig = {},
+    toSupport: SupportConfig = {}
+  ): Promise<void> {
+    const lockedMutex = [];
 
     try {
-      if (! _.isEmpty(toImport.userMappings) && await mutexes.userMappings.lock()) {
-        lockedMutexes.push(mutexes.userMappings);
-        await this.internalIndex.updateMapping('users', toImport.userMappings);
-        await this.internalIndex.refreshCollection('users');
-        this.log.info('[✔] User mappings import successful');
-      }
+      for (const [type, importMethod] of Object.entries(this.importTypes)) {
+        const mutex = new Mutex(`backend:import:${type}`, { timeout: 0 });
+        const isInitialized = await this.ask('core:cache:internal:get', `${BACKEND_IMPORT_KEY}:${type}`);
```

Comment:
Nitpicking: Usually I'm not fond of the `isXXXX` notation for boolean variable. Since they are boolean they already express a state without having to write `is`. This is particularly valid in Typescript since we have strong types.